src/Domain/InstalmentsGenerator.cs

```diff
@@ -1,65 +1,77 @@
-﻿using SFA.DAS.Funding.ApprenticeshipEarnings.Domain.ApprenticeshipFunding;
+﻿using SFA.DAS.Funding.ApprenticeshipEarnings.Domain.Apprenticeship;
+using SFA.DAS.Funding.ApprenticeshipEarnings.Domain.ApprenticeshipFunding;
 
 namespace SFA.DAS.Funding.ApprenticeshipEarnings.Domain;
 
-public interface IInstalmentsGenerator
+public static class InstalmentsGenerator
 {
-    public List<Earning> Generate(decimal total, DateTime startDate, DateTime endDate);
-    public List<Earning> Recalculate(decimal newTotalPrice, DateTime priceDateChange, DateTime endDate, List<Earning> existingEarnings);
-}
-
-public class InstalmentsGenerator : IInstalmentsGenerator
-{
-    public List<Earning> Generate(decimal total, DateTime startDate, DateTime endDate)
+    public static List<Earning> GenerateEarningsForEpisodePrices(List<Price> prices, out decimal onProgramTotal, out decimal completionPayment)
     {
-        var installments = new List<Earning>();
-
-        var startDateMonth = new DateTime(startDate.Year, startDate.Month, 1);
-        var endDateMonth = GetLastPaymentMonth(endDate);
+        var earnings = new List<Earning>();
+        onProgramTotal = 0;
+        completionPayment = 0;
+        var apprenticeshipEndDate = prices.Max(x => x.EndDate);
 
-        while (startDateMonth <= endDateMonth)
+        foreach (var price in prices)
         {
-            installments.Add(new Earning
-            {
-                DeliveryPeriod = startDateMonth.ToDeliveryPeriod(),
-                AcademicYear = startDateMonth.ToAcademicYear()
-            });
-            startDateMonth = startDateMonth.AddMonths(1);
-        }
-
-        var installmentAmount = total / installments.Count;
+            var apprenticeshipFunding = new ApprenticeshipFunding.ApprenticeshipFunding(
+                price.AgreedPrice,
+                price.FundingBandMaximum);
+            onProgramTotal = apprenticeshipFunding.OnProgramTotal;
```

Comment:
Why do we set this value in a loop?

Reply:
And the prices will always be in order oldest to newest so that the onTotalPrice thats applied to the out parameter is the newest?